base/deprecated.jl

```diff
@@ -687,6 +687,20 @@ end
 # After deprecation is removed, enable the @testset "indexing by Bool values" in test/arrayops.jl
 # Also un-comment the new definition in base/indices.jl
 
+# Broadcast no longer defaults to treating its arguments as scalar (#)
+function Broadcast.broadcastable(x)
+    if Base.Broadcast.BroadcastStyle(typeof(x)) isa Broadcast.Unknown
+        depwarn("""
+            broadcast will default to iterating over its arguments in the future. Wrap arguments of
+            type `x::$(typeof(x))` with `Ref(x)` to ensure they broadcast as "scalar" elements.
```

Comment:
Maybe add:

> or define `broadcastable(x::MyType) = Ref(x)` for the conversion to be automatic.

Reply:
I'm guessing that the most common reader of this deprecation will be an end-user.